config/v1/types_authentication.go

```diff
@@ -159,3 +170,74 @@ const (
 	// KubeConfigKey is the key for the kube config file data in a secret
 	KubeConfigKey = "kubeConfig"
 )
+
+type OIDCProvider struct {
+	// Issuer describes atributes of the OIDC token issuer
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	Issuer TokenIssuer `json:"issuer"`
+
+	// ClaimMappings describes rules on how to transform information from an
+	// ID token into a cluster identity
+	ClaimMappings TokenClaimMappings `json:"claimMappings"`
+}
+
+type TokenIssuer struct {
+	// URL is the serving URL of the token issuer.
+	// Must use the https:// scheme.
+	//
+	// +kubebuilder:validation:Required
+	// +required
+	URL string `json:"url"`
+
+	// Audiences is an array of audiences that the token was issued for.
+	// Valid tokens must include at least one of these values in their
+	// "aud" claim.
+	// Must be set to exactly one value.
+	//
+	// +listType=set
+	// +kubebuilder:validation:Required
```

Comment:
can I have one item and be "" or is that invalid?  if invalid, we'll have to work out how to enforce.

Reply:
Added `TokenAudiences` type with MinLength restriction